s3/src/main/java/com/yahoo/ycsb/db/S3Client.java

```diff
@@ -0,0 +1,305 @@
+/**
+ * S3 storage client binding for YCSB.
+ *
+ * Submitted by Ivan Baldinotti on 14/07/2015
+ *
+ */
+package com.yahoo.ycsb.db;
+
+import java.util.HashMap;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.util.*;
+import java.io.IOException;
+
+import com.yahoo.ycsb.ByteArrayByteIterator;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.*;
+import com.amazonaws.auth.*;
+import com.amazonaws.services.s3.model.ObjectMetadata;
+import com.amazonaws.services.s3.model.PutObjectResult;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.regions.Region;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.s3.model.DeleteObjectRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+
+/**
+ * S3 Storage client for YCSB framework.
+ * 
+ * Properties to set:
+ * 
+ * s3.accessKeyId=access key S3 aws
+ * s3.secretKey=secret key S3 aws
+ * s3.endPoint=s3.amazonaws.com
+ * s3.region=us-east-1
+ *
+ * @author ivanB1975
+ */
+public class S3Client extends DB {
+ 
+	private static String key;
+    	private static String bucket;
+	private static String accessKeyId;
+	private static String secretKey;
+	private static String endPoint;
+	private static String region;
+	private static BasicAWSCredentials s3Credentials;
+	private static AmazonS3Client s3Client;
+	private static ClientConfiguration clientConfig;
+
+	/**
+	* Cleanup any state for this storage.
+	* Called once per S3 instance; there is one S3 instance per client thread.
+	*/
+	@Override
+	public void cleanup() throws DBException {
+		try {
+		//this.s3Client.shutdown(); //this should not be used
```

Comment:
I eliminated the shutdown method since it is not supposed to be used normally. Citing the SDK documentation: Shuts down this client object, releasing any resources that might be held open. This is an optional method, and callers are not expected to call it, but can if they want to explicitly release any open resources. Once a client has been shutdown, it should not be used to make any more requests.